python/cudf/cudf/core/column/column.py

```diff
@@ -1911,20 +1911,17 @@ def as_column(
         if dtype is not None:
             column = column.astype(dtype)
         return column
-    elif isinstance(arbitrary, ColumnBase):
-        if dtype is not None:
-            return arbitrary.astype(dtype)
+    elif isinstance(arbitrary, (ColumnBase, cudf.Series, cudf.BaseIndex)):
+        if isinstance(arbitrary, cudf.Series):
+            column = arbitrary._column
+        elif isinstance(arbitrary, cudf.BaseIndex):
+            column = arbitrary._values
         else:
-            return arbitrary
-    elif isinstance(arbitrary, cudf.Series):
-        data = arbitrary._column
-        if dtype is not None:
-            data = data.astype(dtype)
-    elif isinstance(arbitrary, cudf.BaseIndex):
-        data = arbitrary._values
+            column = arbitrary
```

Comment:
You could remove the else clause if you reused the name `arbitrary`. Up to you if you like that, just a suggestion.

Reply:
Sure thing. Changed